### PR TITLE
Convert `CameraRecording` to TS and add stories

### PR DESCRIPTION
**Pull request by haotongwang**
## Description

Title

## Screenshots

image

image

image

image

image

## Steps to Test

1. Run `yarn run storybook` in dashboard `client`

### CODE REVIEW

**Review by Blake-Haydon — DISMISSED** [review]
LGTM! This is really easy to read, nice job. 🎉

**Comment by haotongwang** [comment]
I've merged master but there were some weird type errors. Not sure if my branch was outdated or if the reviews missed something.

@hallgchris In particular it seems like the hooks for camera status was typed as containing the payload information for both devices, which seemed to be different from what we had before (the hooks take a `device` parameter to specify the queried device rather than requesting all the devices' status).

File: client/src/api/v2/camera.ts [hunk 1]
```diff
@@ -3,14 +3,14 @@ import {
   Array,
   Boolean,
   Literal,
-  Null,
+
   Number,
-  Partial,
+
```
Comment (hallgchris) [inline client/src/api/v2/camera.ts]: Is there a reason for keeping these lines?

File: client/src/components/BoostConfigList.tsx [hunk 2]
```diff
@@ -2,16 +2,21 @@ import { faFile, faTrashAlt } from '@fortawesome/free-solid-svg-icons';
...
   config: BoostConfig;
   onSelectConfig: (configName: string) => void;
   onDeleteConfig: (configName: string) => void;
 }
 
+/**
+ * Selectors for BOOST config
+ *
+ * @returns Component
+ */
```
Comment (hallgchris) [inline client/src/components/BoostConfigList.tsx:19]: Whoops, thanks for this

File: client/src/components/v2/CameraRecording.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    primary: {
+        online: false,
+        statusFormatted: null,
+        ip: '192.168.100.97'
+    },
+    secondary: {
+        online: false,
+        statusFormatted: null,
+        ip: '192.168.100.97'
+    },
```
Comment (hallgchris) [inline client/src/components/v2/CameraRecording.stories.tsx]: This is repeated lots, can you extract it to a variable?

File: client/src/components/v2/CameraRecording.tsx [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+            </Card.Subtitle>
+            {/* Fields */}
+            <CameraRecordingStatus statusFormatted={primary?.statusFormatted} />
+          </Col>
+          {/* Device recording controls UI for secondary device */}
+          <Col className={styles.col} key='secondary' sm={6}>
+            <Card.Subtitle className={styles.subtitle}>
+              {/* Name */}
+              <span className={styles.deviceName}>
+                Primary
```
Comment (hallgchris) [inline client/src/components/v2/CameraRecording.tsx]: Should be "Seconday" haha

File: client/src/components/v2/CameraRecording.tsx [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+                {primary.online ? 'Online' : 'Offline'}
+              </Badge>
+              {/* IP */}
+              <div className={styles.push}>{primary.ip}</div>
+            </Card.Subtitle>
+            {/* Fields */}
+            <CameraRecordingStatus statusFormatted={primary?.statusFormatted} />
+          </Col>
+          {/* Device recording controls UI for secondary device */}
+          <Col className={styles.col} key='secondary' sm={6}>
```
Comment (hallgchris) [inline client/src/components/v2/CameraRecording.tsx]: Reckon you could cut down this repeated code? Maybe put the `Col` inside a `['pirmary', 'secondary'].forEach(...)`

File: client/src/components/v2/CameraRecording.tsx [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  startRecording: () => void
+  stopRecording: () => void
+}
+
+/**
+ * Camera recording buttons for starting and stopping video recording.
+ *
+ * Starts/stops recording for both displays.
+ *
+ * This is a feature intended for V3 but is currently in V2 for testing.
```
Comment (hallgchris) [inline client/src/components/v2/CameraRecording.tsx]: I'd argue it's intended for both! Both V2 and V3 camera systems will run the same code and can both record.

File: client/src/components/v2/CameraRecording.tsx [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+export type CameraRecordingPropT = VideoFeedStatus & CameraRecordingStatusProps & { ip: string };
+
+export interface CameraRecordingProps {
+  /** Information for primary device */
+  primary: CameraRecordingPropT
+  /** Information for secondary device */
+  secondary: CameraRecordingPropT
+  startRecording: () => void
+  stopRecording: () => void
```
Comment (hallgchris) [inline client/src/components/v2/CameraRecording.tsx:18]: May as well document these callbacks too

File: client/src/components/v2/CameraRecording.stories.tsx [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+    primary: {
+        online: true,
+        statusFormatted: [
+            {
+                name: "Recording",
+                value: "Off"
+            },
+            {
+                name: "Disk space remaining",
+                value: "1.0 Gb"
```
Comment (hallgchris) [inline client/src/components/v2/CameraRecording.stories.tsx]: Biggest nitpick ever - our actual code uses gibibytes, not gigabits

File: client/src/api/v2/camera.ts [hunk 9]
```diff
@@ -240,10 +240,10 @@ function createStatusPayloadHook<T, U>(
...
 }
 
-const CameraRecordingStatusPayloads = Partial({
-  primary: CameraRecordingStatusPayload,
-  secondary: CameraRecordingStatusPayload,
-});
+// const CameraRecordingStatusPayloads = Partial({
+//   primary: CameraRecordingStatusPayload,
+//   secondary: CameraRecordingStatusPayload,
+// });
```
Comment (hallgchris) [inline client/src/api/v2/camera.ts]: If we no longer need this code we should remove it entirely :)

**Review by hallgchris — CHANGES_REQUESTED** [review]
A few comments! This runs fine for me - what type errors are you getting?

File: client/src/api/v2/camera.ts (hunk 9 above)
Comment (haotongwang) [reply client/src/api/v2/camera.ts]: I wasn't sure if the hooks were intended to return the status payloads for both devices or just for the specified device. The latter is consistent with how it was using in `CameraRecordingStatus` but I vaguely recall discussing the former.

Comment (hallgchris) [reply client/src/api/v2/camera.ts]: Do whatever you think is neatest :) I don't think I have a big investment in either (although I haven't had much of a dig through the code recently)

File: client/src/api/v2/camera.ts (hunk 1 above)
Comment (haotongwang) [reply client/src/api/v2/camera.ts]: Nope, my autoformatter just didn't remove them :/
5959e5a

File: client/src/components/v2/CameraRecording.stories.tsx (hunk 3 above)
Comment (haotongwang) [reply client/src/components/v2/CameraRecording.stories.tsx]: done
14d165d

File: client/src/components/v2/CameraRecording.tsx (hunk 4 above)
Comment (haotongwang) [reply client/src/components/v2/CameraRecording.tsx]: done
e805a93

File: client/src/components/v2/CameraRecording.tsx (hunk 5 above)
Comment (haotongwang) [reply client/src/components/v2/CameraRecording.tsx]: done but i'm not sure if this is that much better than what we had before, it does reduce the repeated code but increases the complexity.
e805a93

File: client/src/components/v2/CameraRecording.tsx (hunk 6 above)
Comment (haotongwang) [reply client/src/components/v2/CameraRecording.tsx]: Done
b4b461d

File: client/src/components/v2/CameraRecording.tsx (hunk 7 above)
Comment (haotongwang) [reply client/src/components/v2/CameraRecording.tsx:18]: Done

File: client/src/components/v2/CameraRecording.stories.tsx (hunk 8 above)
Comment (haotongwang) [reply client/src/components/v2/CameraRecording.stories.tsx]: done
97a7cd0

File: client/src/api/v2/camera.ts (hunk 9 above)
Comment (haotongwang) [reply client/src/api/v2/camera.ts]: done
c294012

File: client/src/components/v2/CameraRecording.tsx (hunk 5 above)
Comment (hallgchris) [reply client/src/components/v2/CameraRecording.tsx]: Hmm I didn't realise we'd have to do the `device`/`deviceName` thing, but I reckon it's still worthwhile! Makes it clear that the two columns should behave identically.

**Review by hallgchris — APPROVED** [review]
LGTM!

**Pull request merged**